base-template.sh

```diff
@@ -758,10 +758,16 @@ show_prompt() {
         # else: Running fallback...
     fi
 
-    # Read from stdin
+    # Try to read from `dev/tty` if available.
+    # otherwise just read from stdin.
     printf "%s %s [%s]:" "$TEXT" "$HINT_TEXT" "$SHORT_OPTIONS"
-    # shellcheck disable=SC2229
-    read -r "$VARIABLE" </dev/tty
+    if [ -f /dev/tty ]; then
```

Comment:
I assume you tested this works?
I'm a bit cautious whether the `-f` test is reliable for a non-regular file, but happy to go with it if it works.

As an alternative, don't we set some Git variable when installing on the server side, where you typically won't have tty?

Reply:
Test works, yes. I thought so too, but we can only test about bare or non bare which is a bit vague, since it does not classify only for server-side repos.
Youu are probably right: `-f ` is probably not working, at least it doesnt on alpine in the terminal
I change to  `if [ -t 0]` instead which is the correct way I think

Reply:
Thanks for looking into it!
Not sure whether `[ -t 0 ]` is portable, I think I read it's not supported somewhere, but could be wrong.
We could also try `[ -e /dev/tty ]` maybe? Not sure if that wants files or if it's fine if it's anything. 🤷‍♂️
And if all else fails, something like `ls /dev/tty` could maybe work?
I'm not really sure, so I'd let testing to you if you get a chance. :)

Reply:
It is POSIX isn't it: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html

Reply:
Yeah, it does look like it. Thanks for checking!